cli/autoconfig/autoconfig.go

```diff
@@ -11,45 +11,70 @@ import (
 	cache  = flag.Bool("cache", false, "If true, perform build with remote caching.")
 	dev    = flag.Bool("dev", false, "If true, point at buildbuddy dev cluster.")
+	apiKey = flag.String("api_key", "", "API key to use for BuildBuddy services.")
 )
 
-func Configure(bazelFlags *commandline.BazelFlags, filteredOSArgs []string) (*commandline.BazelFlags, []string) {
+type BazelOpts struct {
+	APIKey             string
+	BuildBuddyEndpoint string
+	EnableRemoteBazel  bool
+}
+
+func Configure(bazelFlags *commandline.BazelFlags, filteredOSArgs []string) (*commandline.BazelFlags, *BazelOpts, []string) {
 	serviceDomain := "buildbuddy.io"
 	if *dev {
 		serviceDomain = "buildbuddy.dev"
 	}
 
 	if *events && bazelFlags.BESBackend == "" {
-		bazelFlags.BESBackend = fmt.Sprintf("grpcs://cloud.%s", serviceDomain)
+		bazelFlags.BESBackend = fmt.Sprintf("grpcs://remote.%s", serviceDomain)
 		filteredOSArgs = append(filteredOSArgs, fmt.Sprintf("--bes_results_url=https://app.%s/invocation/", serviceDomain))
 	}
 
-	if *remote && bazelFlags.RemoteExecutor == "" {
-		bazelFlags.RemoteExecutor = fmt.Sprintf("grpcs://cloud.%s", serviceDomain)
-		filteredOSArgs = append(filteredOSArgs, "--crosstool_top=@buildbuddy_toolchain//:toolchain")
-		filteredOSArgs = append(filteredOSArgs, "--javabase=@buildbuddy_toolchain//:javabase_jdk8")
-		filteredOSArgs = append(filteredOSArgs, "--host_javabase=@buildbuddy_toolchain//:javabase_jdk8")
-		filteredOSArgs = append(filteredOSArgs, "--java_toolchain=@buildbuddy_toolchain//:toolchain_jdk8")
-		filteredOSArgs = append(filteredOSArgs, "--host_java_toolchain=@buildbuddy_toolchain//:toolchain_jdk8")
-		filteredOSArgs = append(filteredOSArgs, "--host_platform=@buildbuddy_toolchain//:platform")
-		filteredOSArgs = append(filteredOSArgs, "--platforms=@buildbuddy_toolchain//:platform")
-		filteredOSArgs = append(filteredOSArgs, "--extra_execution_platforms=@buildbuddy_toolchain//:platform")
+	if (*remote == remoteExec || *remote == remoteBazel) && bazelFlags.RemoteExecutor == "" {
+		bazelFlags.RemoteExecutor = fmt.Sprintf("grpcs://remote.%s", serviceDomain)
+		if *remote == remoteExec {
+			filteredOSArgs = append(filteredOSArgs, "--crosstool_top=@buildbuddy_toolchain//:toolchain")
+			filteredOSArgs = append(filteredOSArgs, "--javabase=@buildbuddy_toolchain//:javabase_jdk8")
+			filteredOSArgs = append(filteredOSArgs, "--host_javabase=@buildbuddy_toolchain//:javabase_jdk8")
+			filteredOSArgs = append(filteredOSArgs, "--java_toolchain=@buildbuddy_toolchain//:toolchain_jdk8")
+			filteredOSArgs = append(filteredOSArgs, "--host_java_toolchain=@buildbuddy_toolchain//:toolchain_jdk8")
+			filteredOSArgs = append(filteredOSArgs, "--host_platform=@buildbuddy_toolchain//:platform")
+			filteredOSArgs = append(filteredOSArgs, "--platforms=@buildbuddy_toolchain//:platform")
+			filteredOSArgs = append(filteredOSArgs, "--extra_execution_platforms=@buildbuddy_toolchain//:platform")
```

Comment:
can you have two functions in this file that return a []string?
```
func bbToolchainFlags() []string {
  return []string{
    "--crosstool_top=@buildbuddy_toolchain//:toolchain",
    ...
  }
}

func remoteBazelFlags() []string {
  ...
}
```

Then your code can be:

```
if *remote == remoteExec {
  filteredOSArgs = append(filteredOSArgs, bbToolchainFlags()...)
} else {
  filteredOSArgs = append(filteredOSArgs, remoteBazelFlags()...)
}
```